astropy/config/configuration.py

```diff
@@ -682,148 +665,70 @@ def reload_config(packageormod=None, rootname=None):
-      Astropy 0.4, when APE3 was implemented and the config file
-      contained commented-out values by default.
+    actually contain any meaningful content, i.e. if it contains only comments
+    or is completely empty.
     """
-    # We want to calculate the md5sum using universal line endings, so
-    # that even if the files had their line endings converted to \r\n
-    # on Windows, this will still work.
-
-    content = content.encode('latin-1')
-
-    # The jquery_url setting, present in 0.3.2 and later only, is
-    # effectively auto-generated by the build system, so we need to
-    # ignore it in the md5sum calculation for 0.3.2.
-    content = re.sub(br'\njquery_url\s*=\s*[^\n]+', b'', content)
-
-    # First determine if the config file has any effective content
-    buffer = io.BytesIO(content)
-    buffer.seek(0)
+    buffer = io.StringIO(content)
     raw_cfg = configobj.ConfigObj(buffer, interpolation=True)
-    for v in raw_cfg.values():
-        if len(v):
-            break
-    else:
-        return True
+    # If any of the items is set, return False
+    return not any(len(v) > 0 for v in raw_cfg.values())
 
-    # Now determine if it matches the md5sum of a known, unedited
-    # config file.
-    known_configs = set([
-        '7d4b4f1120304b286d71f205975b1286',  # v0.3.2
-        '5df7e409425e5bfe7ed041513fda3288',  # v0.3
-        '8355f99a01b3bdfd8761ef45d5d8b7e5',  # v0.2
-        '4ea5a84de146dc3fcea2a5b93735e634'   # v0.2.1, v0.2.2, v0.2.3, v0.2.4, v0.2.5
-    ])
 
-    md5 = hashlib.md5()
-    md5.update(content)
-    digest = md5.hexdigest()
-    return digest in known_configs
-
-
-# this is not in __all__ because it's not intended that a user uses it
-def update_default_config(pkg, default_cfg_dir_or_fn, version=None, rootname='astropy'):
+def create_config_file(pkg, rootname='astropy', force=False):
     """
-    Checks if the configuration file for the specified package exists,
-    and if not, copy over the default configuration.  If the
-    configuration file looks like it has already been edited, we do
-    not write over it, but instead write a file alongside it named
-    ``pkg.version.cfg`` as a "template" for the user.
+    Create the default configuration file for the specified package.
+    If the file already exists, it is updated only if it has not been
+    modified.  Otherwise the ``force`` flag is needed to overwrite it.
 
     Parameters
     ----------
     pkg : str
         The package to be updated.
-    default_cfg_dir_or_fn : str
-        The filename or directory name where the default configuration file is.
-        If a directory name, ``'pkg.cfg'`` will be used in that directory.
-    version : str, optional
-        The current version of the given package.  If not provided, it will
-        be obtained from ``pkg.__version__``.
     rootname : str
         Name of the root configuration directory.
+    force : bool
```

Comment:
More consistent with our existing API over at unified IO?
```suggestion
    overwrite : bool
```

Reply:
Done.